File: packages/sdk-python/steno_ai/keys.py
"""Key client — manage API keys and usage."""

from .client import HttpClient


class KeyClient:
    def __init__(self, http: HttpClient):
        self._http = http

    def create(self, name: str) -> dict:
        """Create a new API key."""
        return self._http.request("POST", "/v1/keys", json={"name": name})

    def list(self) -> list:
        """List all API keys."""
        return self._http.request("GET", "/v1/keys")

    def revoke(self, key_id: str) -> None:
        """Revoke an API key."""
        self._http.request("DELETE", f"/v1/keys/{key_id}")

    def usage(self) -> dict:
        """Get usage stats for the current key."""
        return self._http.request("GET", "/v1/usage")

File: packages/sdk-ts/src/clients/keys.ts
import type { HttpClient } from '../http.js';
import type { ApiKey, UsageResponse } from '../types.js';

export class KeyClient {
  constructor(private readonly http: HttpClient) {}

  /** Create a new API key. */
  async create(name: string): Promise<ApiKey & { key: string }> {
    return this.http.request('POST', '/v1/keys', { name });
  }

  /** List all API keys. */
  async list(): Promise<ApiKey[]> {
    return this.http.request('GET', '/v1/keys');
  }

  /** Revoke an API key. */
  async revoke(keyId: string): Promise<void> {
    return this.http.request('DELETE', `/v1/keys/${keyId}`);
  }

  /** Get usage stats for the current key. */
  async usage(): Promise<UsageResponse> {
    return this.http.request('GET', '/v1/usage');
  }
}
